packages/svelte/src/internal/client/dom/elements/bindings/select.js

```diff
@@ -91,6 +91,28 @@ export function bind_select_value(select, get_value, update) {
 		select.__value = value;
 		mounting = false;
 	});
+
+	// If one of the options gets removed from the DOM, the value might have changed
+	effect(() => {
+		var observer = new MutationObserver(() => {
+			// @ts-ignore
+			var value = select.__value;
+			select_option(select, value, mounting);
+			/** @type {HTMLOptionElement | null} */
+			var selected_option = select.querySelector(':checked');
+			if (selected_option === null || get_option_value(selected_option) !== value) {
+				update('');
+			}
```

Comment:
This does not take `<select multiple ...>` into account. If three options are selected and one is removed, the remaining two should still be selected ([Example](https://svelte-5-preview-a5rk26gw7-svelte.vercel.app/#H4sIAAAAAAAACq2QwW6EIBBAf4XQJrqJqXWPFk36DT2ue0AcW7KIRMZNGsK_N6C2W9v01AuZGebBm3G0lwosLU-Oaj4ALemzMTSj-G5CYq-gEGhG7ThPIlSYFZM0WDe6QQVIJMJgSUXuLXKE9JTwIslIwo_hbGPcHpPz4anRGzLMCqVR8AIKBEL3B71wkbJSv_7GBCR0sfzLTLN2Rhw1GbVQUlwqlx5IVS-yD9YoKSB9zEhx8PUEA-EFyxdioU29ORK7flgSt_f2LDd1o9nS8jkWaaXuyitXM1Q_mbg4dwdcvK274zYGPlw0yEaDctS1izWWr2mk8kD5OGp8bpPtFcdb0e-r2mne2u0a_9WNZnQYO9lL6GiJ0wz-7D8A765QpW8CAAA=)).